cli/notify/notify.go

```diff
@@ -0,0 +1,121 @@
+package notify
+
+/**
+ * urfave/cli Notify "Extension"
+ *
+ * This sub-package of rig is intended for potential separation into a
+ * standalone package or PR to the main framework. As such, it should not have
+ * any built-in assumptions or identifiers about Rig, nor depend on anything in
+ * the rig codebase.
+ */
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/martinlindhe/notify"
+	"github.com/urfave/cli"
+)
+
+var config *NotifyConfig
+
+type NotifyConfig struct {
+	// Holds a list of command names and their default value for notification purposes.
+	// This needs to be retained after setting up the flags to more easily determine
+	// the notification criteria for each command. Without this, there would be a
+	// lot more array iterating.
+	Commands map[string]bool
+	// The name of the app for environment variables.
+	Id string
+	// Relative path to notification logo.
+	Icon string
+	// The label for the app to be displayed in notifications.
+	Label string
+}
+
+// Initialize notifications.
+func Init(id string, label string, icon string, commands map[string]bool) error {
```

Comment:
I'm thinking that it likely makes sense to make Notify a struct/object since it has config that is uses and such. Kind of like Commands themselves.